File: server/router/normal/gost_client_tunnel.go
package normal

import (
	"github.com/gin-gonic/gin"
	"server/controller/normal/gost_client_tunnel"
	"server/global"
	"server/router/middleware"
)

func InitGostClientTunnel(group *gin.RouterGroup) {
	g := group.Group("gost/client/tunnel", middleware.Auth(global.Jwt))
	g.POST("create", gost_client_tunnel.Create)
	g.POST("renew", gost_client_tunnel.Renew)
	g.POST("update", gost_client_tunnel.Update)
	g.POST("enable", gost_client_tunnel.Enable)
	g.POST("delete", gost_client_tunnel.Delete)
	g.POST("page", gost_client_tunnel.Page)
}

File: web/src/api/normal/gost_client_tunnel.js
import request from "../../request/index.js";

const baseUrl = '/v1/normal/gost/client/tunnel'

export const apiNormalGostClientTunnelList = () => {
    return request.request({
        url: `${baseUrl}/list`,
        method: 'POST',
    })
}

export const apiNormalGostClientTunnelPage = (data) => {
    return request.request({
        url: `${baseUrl}/page`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientTunnelCreate = (data) => {
    return request.request({
        url: `${baseUrl}/create`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientTunnelRenew = (data) => {
    return request.request({
        url: `${baseUrl}/renew`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientTunnelMigrate = (data) => {
    return request.request({
        url: `${baseUrl}/migrate`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientTunnelEnable = (data) => {
    return request.request({
        url: `${baseUrl}/enable`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientTunnelUpdate = (data) => {
    return request.request({
        url: `${baseUrl}/update`,
        method: 'POST',
        data
    })
}


export const apiNormalGostClientTunnelDelete = (data) => {
    return request.request({
        url: `${baseUrl}/delete`,
        method: 'POST',
        data
    })
}